chore: Clean up get_extra_menu_items methods & Test for djangocms-transfer's PluginImporter

mrbazzan opened the pull request:
## Description
- Extensive test for PluginImporter.import_plugin_views

## Checklist

<!--
Please check the following items before submitting, otherwise,
your pull request will be closed.
Use 'x' to check each item: [x] I have ...
-->

* [x] I have opened this pull request against ``master``
* [ ] I have added or modified the tests when changing logic
* [ ] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog
* [ ] I have read the [contribution guidelines ](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst) and I have joined #workgroup-pr-review on 
[Discord](https://www.django-cms.org/discord) to find a “pr review buddy” who is going to review my pull request.

## Summary by Sourcery

Extract common logic for building plugin and placeholder export/import menu items into a new static helper and remove the deprecated compatibility method

Enhancements:
- Removed the deprecated get_extra_global_plugin_menu_items method
- Introduced a private _get_extra_menu_items helper to DRY up URL parameter encoding and menu item construction
- Updated get_extra_plugin_menu_items and get_extra_placeholder_menu_items to delegate to the new helper

mrbazzan committed:
chore: Clean up get_extra_menu_items methods

mrbazzan committed:
tests: Simplify query by using page_content

mrbazzan committed:
chore: Optimize plugin validation in forms.py

mrbazzan committed:
tests: Test for extra menu items

mrbazzan committed:
tests: Test for plugin import view

mrbazzan commented on djangocms_transfer/cms_plugins.py:
```diff
@@ -33,21 +33,12 @@ def get_plugin_urls(self):
-        )
+    @staticmethod
+    def _get_extra_menu_items(query_params, request):
+        data = urlencode({
+            "language": get_language_from_request(request),
+            **query_params
```
@sourcery-ai
we control `query_param` so it can't contain unintended key right?

mrbazzan commented on tests/test_plugins.py:
```diff
@@ -0,0 +1,182 @@
+                "import_file": SimpleUploadedFile("file.txt", b"[null, null]")
+            }
+            with self.assertRaises(IndexError):
+                self.client.post(path=request_path, data=post_data)
```
@fsbraun 
I don't this is a plugin-specific test. The error is raised from `get_plugin_tree`
helper method

fsbraun replied:
@mrbazzan The error in the test_plugins test case actually revealed a real bug, I believe: I pushed a fix here: https://github.com/django-cms/djangocms-transfer/pull/50/commits/ad1799510bf3a7c7b84b5bf9ca0a125eede83b23 

So, the tests already created value! Nice!

For it to work with django-cms 4.1, we need compatibility shim for `get_plugin_tree` which was not there yet.

mrbazzan commented on tests/test_plugins.py:
```diff
@@ -0,0 +1,182 @@
+                "import_file": SimpleUploadedFile("file.txt", b"[null, null]")
+            }
+            with self.assertRaises(IndexError):
+                self.client.post(path=request_path, data=post_data)
```
Things one too

fsbraun committed:
fix: Plugin validation in forms.py failed on CMSPlugin class (which is always bound)

fsbraun committed:
chore: Add CMS 4.1 compatibility shim

fsbraun committed:
fix: ruff issues

fsbraun reviewed (commented):
@mrbazzan Looks good to me!

The only question I have is why the tests are inside the `transfer` folder. Are you planning for another test folder. If it will only be one folder with tests, I'd rather have the tests in the `tests` folder.

mrbazzan committed:
chore: Remove duplicate json import

mrbazzan committed:
chore: Move the tests from the transfer/ folder

mrbazzan commented:
> The only question I have is why the tests are inside the `transfer` folder. Are you planning for another test folder. If it will only be one folder with tests, I'd rather have the tests in the `tests` folder.

@fsbraun 
I didn't create the `transfer` folder; it was already in the folder, I just added the new files to it.
I have removed it now. Ready for review

mrbazzan commented on tests/test_plugins.py:
```diff
@@ -0,0 +1,182 @@
+
+        with self.login_user_context(self.user):
+            # GET plugin import
+            response = self.client.get(
```
@sourcery-ai extract-method

mrbazzan replied:
I don't agree that this method is cleaner
- a function just for one/two line of codes? I fancy the code the way it is

mrbazzan committed:
tests: Test for export plugin views

mrbazzan commented:
@fsbraun 
Ready for review

fsbraun reviewed (approved):
Excellent work! Thank you so much!
